pkg/staticnetworkconfig/generator_test.go

```diff
@@ -278,6 +278,19 @@ parent=eth1
 			})
 			Expect(err).ToNot(HaveOccurred())
 		})
+		It("vlan with underlying interface - with mapping without parent", func() {
+			err := staticNetworkGenerator.validateInterfaceNamesExistence(models.MacInterfaceMap{
```

Comment:
I think the unit test should test `ValidateStaticConfigParams` (exported method) and not `validateInterfaceNamesExistence`. In that way the whole process of generating the connection files from nmstate yamls and validating them will be covered.
I would also add a similar test for bond (since it is not straight forward to understand why vlan requires as special condition and bond not).